crates/sui-indexer/src/handlers/objects_snapshot_processor.rs

```diff
@@ -221,26 +212,43 @@ impl ObjectsSnapshotProcessor {
                         .await?
                         .unwrap_or_default();
 
+                    // Fetch new changes if buffer is empty
+                    if buffered_changes.is_empty() {
+                        if let Some(new_changes) = stream.next().await {
+                            buffered_changes.extend(new_changes);
+                            // Can't really guarantee order, so we sort checkpoints ascending here
+                            buffered_changes.sort_by_key(|change| change.checkpoint_sequence_number);
+                        }
+                    }
+
                     // We update the snapshot table when it falls behind the rest of the indexer by more than the min_lag.
-                    while latest_indexer_cp >= start_cp + config.snapshot_min_lag as u64 {
-                        // Stream the next object changes to be committed to the store.
-                        if let Some(object_changes_batch) = stream.next().await {
-                            let first_checkpoint_seq = object_changes_batch.first().as_ref().unwrap().checkpoint_sequence_number;
-                            let last_checkpoint_seq = object_changes_batch.last().as_ref().unwrap().checkpoint_sequence_number;
-                            info!("Objects snapshot processor is updating objects snapshot table from {} to {}", first_checkpoint_seq, last_checkpoint_seq);
-
-                            let changes_to_commit = object_changes_batch.into_iter().map(|obj| obj.object_changes).collect();
-                            store.backfill_objects_snapshot(changes_to_commit)
-                                .await
-                                .unwrap_or_else(|_| panic!("Failed to backfill objects snapshot from {} to {}", first_checkpoint_seq, last_checkpoint_seq));
-                            start_cp = last_checkpoint_seq + 1;
-
-                            // Tells the package buffer that this checkpoint has been processed and the corresponding package data can be deleted.
-                            commit_notifier.send(Some(last_checkpoint_seq)).expect("Commit watcher should not be closed");
-                            metrics
-                                .latest_object_snapshot_sequence_number
-                                .set(last_checkpoint_seq as i64);
+                    while latest_indexer_cp >= start_cp + config.lag as u64 && !buffered_changes.is_empty() {
```

Comment:
`>` instead of `>=`? b/c when latest_indexer_cp == start_cp + config.lag, max_allowed_cp == start_cp, and no updates are allowed.

Reply:
I think we have `>=` because of the `start_cp = last_checkpoint_seq + 1;`

so for example, if latest indexer cp is 10, lag is 6, and start_cp is 3, and we've entered the loop, we'll exit with `start_cp=4`. 

we want the >= so we can also index start_cp=4 to `objects_snapshot`

although alternatively, i suppose we can make it `latest_indexer_cp > start_cp + config.lag` and remove the `+1`
@emmazzz do you recall why the processor logic is set up this way?

Reply:
`start_cp = last_checkpoint_seq + 1;` goes together with `for cp in start_cp..=max_allowed_cp`, more specifically start_cp needs to be included to avoid gaps.

Reply:
nvm, now it looks right, there is at least one element in `for cp in start_cp..=max_allowed_cp`.

Reply:
added a comment to elucidate this